# Defined in udev rules, change according to your system. Typically /dev/ttyUSB0
serial_port: /dev/ttyUSB0

# Acceptable baud rates : 9600, 19200, 38400, 57600, 115200, 230400, 460800, 921600
# Datasheet states 128000 works but from experiments it does not.
serial_baud: 921600

# Acceptable data rates in Hz: 1, 2, 4, 5, 10, 20, 25, 40, 50, 100, 200
# Baud rate must be able to handle the data rate
async_output_rate: 40

# Device IMU Rate as set by the manufacturer (800Hz unless specified otherwise) vn300 has 400 hz imu
# This value is used to set the serial data packet rate
fixed_imu_rate: 400

# Frame id to publish data in
frame_id: vectornav

# Frame id for map / odom
map_frame_id: vectornav_map

# namespace for published topics. ns = vectornav => /vetornav/Odom, /vectornav/Imu...
ns: vectornav

# Data publication form, true for East North Up or false for North East Down <- false is default setting
ned_to_enu: false

# baseline is the vector from antenna A to antenna B. This need to be in the ins refrence frame.
baseline: [-2.91, -0.55, 0.04]

# uncertainty in the measurement of the baseline. Vectornav recommends setting it equal to TWICE 
# the worst case. 
baseline_uncertainty: [0.1, 0.1, 0.1]

# Antenna offset is the vector from the ins to antenna A.
antenna_offset: [1.56, 0.31, 0.07]

# Make sure all covariances below are of type xx.xx , i.e. double so that the rpc is parsed correctly

# Linear Acceleration Covariances not produced by the sensor
linear_accel_covariance: [0.01,  0.0,    0.0,
                          0.0,   0.01,   0.0,
                          0.0,   0.0,    0.01]

# Angular Velocity Covariances not produced by the sensor
angular_vel_covariance: [0.01,  0.0,   0.0,
                         0.0,   0.01,  0.0,
                         0.0,   0.0,   0.01]

# Orientation covariance overwritten in driver, this is included just as an extra
orientation_covariance: [0.01,  0.0,   0.0,
                            0.0,   0.01,  0.0,
                            0.0,   0.0,   0.01]
